pytest_rerunfailures.py

```diff
@@ -160,6 +185,10 @@ def pytest_runtest_protocol(item, nextitem):
                     # will rerun test, log intermediate result
                     item.ihook.pytest_runtest_logreport(report=report)
 
+                # cleanin item's cashed results from any level of setups
+                _remove_cached_results_from_fixtures(item)
+                _remove_failed_setup_state_from_session(item)
```

Comment:
What I am not sure about is the effect for the result of a module or session scoped fixture. Could you please add a test for them as well, that they are not affected by this change? I guess this should not re-initiate those fixtures as they could be expensive database fixtures.

Reply:
I would not like to re-initialize all fixtures, but only the one who failed. So that should be the case with the current state of the PR.

Reply:
That is correct. The only failed fixture will be erased from any previous results.